evalml/objectives/objective_base.py

```diff
@@ -42,6 +42,12 @@ def perfect_score(cls):
     def is_bounded_like_percentage(cls):
         """Returns whether this objective is bounded between 0 and 1, inclusive."""
 
+    @property
+    @classmethod
+    @abstractmethod
+    def expected_range(cls):
+        """Returns the expected range of the objective, which is not necessarily the possible ranges."""
```

Comment:
Nit: Could you add a little more detail here? If a user is trying to define their own objective, they may not quite understand what this should be